receiver/mongodbatlasreceiver/access_logs.go

```diff
@@ -0,0 +1,336 @@
+		if err != nil {
+			alr.logger.Error("error retrieving cluster information", zap.Error(err), zap.String("project", pc.Name))
+			return err
+		}
+		filteredClusters, err := filterClusters(clusters, pc)
+		if err != nil {
+			alr.logger.Error("error filtering clusters", zap.Error(err), zap.String("project", pc.Name))
+			return err
+		}
+		for _, cluster := range filteredClusters {
+			alr.pollCluster(ctx, project, cluster, st, et)
+		}
+	}
+
+	alr.nextStartTime = &et
+	return alr.checkpoint(ctx)
+}
+
+func (alr *accessLogsReceiver) pollCluster(ctx context.Context, project *mongodbatlas.Project, cluster mongodbatlas.Cluster, startTime, now time.Time) {
+	opts := &internal.GetAccessLogsOptions{
+		MaxDate:    now,
+		MinDate:    startTime,
+		AuthResult: alr.authResult,
+		NLogs:      maximumLogEntriesPerRequest,
+	}
+
+	fullLogs := make([]*mongodbatlas.AccessLogs, 0)
+	for {
+		accessLogs, err := alr.client.GetAccessLogs(ctx, project.ID, cluster.ID, opts)
+		if err != nil {
+			alr.logger.Error("unable to get access logs", zap.Error(err), zap.String("project", project.Name),
+				zap.String("clusterID", cluster.ID), zap.String("clusterName", cluster.Name))
+			if len(fullLogs) > 0 {
+				alr.logger.Info("sending partial access logs", zap.Int("logs", len(fullLogs)))
+				break
+			}
+			return
+		}
+
+		fullLogs = append(fullLogs, accessLogs...)
+
+		// If we get back less than the maximum number of logs, we can assume that we've retrieved all of the logs for this
+		// time period.
+		if len(accessLogs) < maximumLogEntriesPerRequest {
+			break
+		}
+
+		// If we get back the maximum number of logs, we need to re-query with a new start time. While undocumented, the API
+		// returns the most recent logs first. If we get the maximum number of logs back, we can assume that
+		// there are more logs to be retrieved. We'll re-query with the same start time, but the end
+		// time set to just before the timestamp of the last log entry returned.
+		lastLog := accessLogs[len(accessLogs)-1]
+		body, err := parseLogMessage(lastLog)
+		if err != nil {
+			// If body couldn't be parsed, we'll still use the outer Timestamp field to determine the new max date.
+			body = map[string]interface{}{}
+		}
+		lastLogTimestamp, err := getTimestamp(lastLog, body)
+		if err != nil {
+			alr.logger.Error("unable to parse last log timestamp", zap.Error(err), zap.String("project", project.Name),
+				zap.String("clusterID", cluster.ID), zap.String("clusterName", cluster.Name))
+			break
+		}
+		opts.MaxDate = lastLogTimestamp.Add(-1 * time.Millisecond)
+
+		// If the new max date is before the min date, we've retrieved all of the logs for this time period
+		// and receiving the maximum number of logs back is a coincidence.
+		if opts.MaxDate.Before(opts.MinDate) {
+			break
+		}
+	}
+
+	nowTimestamp := pcommon.NewTimestampFromTime(now)
+	logs := transformAccessLogs(nowTimestamp, fullLogs, project, cluster, alr.logger)
+
+	if logs.LogRecordCount() > 0 {
```

Comment:
I can imagine this being a bottleneck for logs. Say we are getting the max number of log entries of 20000 and we keep adding logs to the fullLogs. We only ever send the logs down at once so we could theoretically stream logs down the telemetry pipeline. Could we try consuming logs after each request when safe to not overload processors with large batches of logs?